src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -176,18 +177,33 @@ public void create(final EventTypeBase eventType, final boolean checkAuth)
             authorizationValidator.validateAuthorization(eventType.asBaseResource());
         }
 
-        eventTypeRepository.saveEventType(eventType);
-
+        final AtomicReference<EventType> createdEventType = new AtomicReference<>(null);
+        final AtomicReference<Timeline> createdTimeline = new AtomicReference<>(null);
         try {
-            timelineService.createDefaultTimeline(eventType,
-                    partitionsCalculator.getBestPartitionsCount(eventType.getDefaultStatistic()));
-        } catch (final Exception e) {
-            try {
-                eventTypeRepository.removeEventType(eventType.getName());
-            } catch (final NoSuchEventTypeException e1) {
-                LOG.error("Error creating event type {}", eventType, e1);
+            transactionTemplate.execute(action -> {
+                createdEventType.set(eventTypeRepository.saveEventType(eventType));
+                createdTimeline.set(timelineService.createDefaultTimeline(eventType,
+                        partitionsCalculator.getBestPartitionsCount(eventType.getDefaultStatistic())));
+                return null;
+            });
+            eventTypeRepository.notifyUpdated(eventType.getName());
+        } catch (final RuntimeException ex) {
+            LOG.error("Failed to create event type, will clean up created data", ex);
+            if (null != createdTimeline.get()) {
+                try {
+                    timelineService.rollbackTopic(createdTimeline.get());
+                } catch (final RuntimeException ex2) {
+                    LOG.warn("Failed to rollback timeline", ex2);
+                }
             }
-            throw e;
+            if (null != createdEventType.get()) {
+                try {
+                    eventTypeRepository.removeEventType(createdEventType.get().getName());
+                } catch (RuntimeException ex2) {
+                    LOG.warn("Failed to rollback event type creation (which is highly expected)", ex2);
+                }
+            }
+            throw ex;
```

Comment:
maybe you want to wrap it in `InternalNakadiException` and give a good description